### PR TITLE
Update ONNX to 1.15.0rc1

### PR DESCRIPTION
**Pull request by liqunfu**
### Description
to use onnx 1.15.0rc1

### Motivation and Context
prepare for 1.17.0 release

### CODE REVIEW

File: cmake/deps.txt [hunk 1]
```diff
@@ -24,7 +24,7 @@ microsoft_gsl;https://github.com/microsoft/GSL/archive/refs/tags/v4.0.0.zip;cf36
 microsoft_wil;https://github.com/microsoft/wil/archive/refs/tags/v1.0.230629.1.zip;e4a542a323c070376f7c2d1973d0f7ddbc1d2fa5
 mimalloc;https://github.com/microsoft/mimalloc/archive/refs/tags/v2.1.1.zip;d5ee7d34223d0567892db5179849939c8769dc41
 mp11;https://github.com/boostorg/mp11/archive/refs/tags/boost-1.82.0.zip;9bc9e01dffb64d9e0773b2e44d2f22c51aace063
-onnx;https://github.com/onnx/onnx/archive/14303de049144035dfd94ace5f7a3b44773b1aad.zip;250eab9690392b248d75b56e605fb49eca373442
+onnx;https://github.com/onnx/onnx/archive/0c296085f9f65f0f8ef7aec7b9eed55faf37dc40.zip;01ca9e955a03a9183e3d278e96f975f1a762cef1
```
Comment (snnn) [inline cmake/deps.txt:27]: Run https://github.com/microsoft/onnxruntime/blob/main/cgmanifests/generate_cgmanifest.py

**Comment by snnn** [comment]
Please also update the cmake submodule, which is still used by web CI pipeline.

**Review by snnn — DISMISSED** [review]
Thank you!

**Review by thiagocrepaldi — DISMISSED** [review]

**Review by jchen351 — DISMISSED** [review]

**Review by faxu — APPROVED** [review]
This will be updated again when onnx 1.15 officially releases, right?

**Review by snnn — APPROVED** [review]

**Pull request merged**